### PR TITLE
Add 'dt_original_post_parent' to post meta

### PR DESCRIPTION
**Pull request by sillybean**
Store the original post_parent ID in post meta to make it easier to restore parent/child relationships after push/pull. See #44

### CODE REVIEW

File: includes/classes/ExternalConnections/WordPressExternalConnection.php [hunk 1]
```diff
@@ -300,6 +300,7 @@ public function pull( $items ) {
 
 			$post_props = get_object_vars( $post );
 			$post_array = array();
+			$original_post_parent = wp_get_post_parent_id( $item_array['post_id'] );
```
Comment (barryceelen) [inline includes/classes/ExternalConnections/WordPressExternalConnection.php]: I _think_ this is not referring to the correct post in this context. The original post is the `remote_post_id`. The WP_Post object for this post is stored in the `$post` variable on line 294, so we could use `$post->post_parent`.

File: includes/classes/ExternalConnections/WordPressExternalConnection.php [hunk 2]
```diff
@@ -335,6 +336,7 @@ public function pull( $items ) {
 			update_post_meta( $new_post, 'dt_original_post_url', esc_url_raw( $post_array['link'] ) );
 			update_post_meta( $new_post, 'dt_original_site_name', sanitize_text_field( $post_array['original_site_name'] ) );
 			update_post_meta( $new_post, 'dt_original_site_url', sanitize_text_field( $post_array['original_site_url'] ) );
+			update_post_meta( $new_post, 'dt_original_post_parent', (int) $original_post_parent );
```
Comment (barryceelen) [inline includes/classes/ExternalConnections/WordPressExternalConnection.php]: A check for `$original_post_parent !== 0` before adding the custom field would be good, so as not to store unnecessary data.  The field not being present would imply that the origin post did not have a parent.

File: includes/classes/InternalConnections/NetworkSiteConnection.php [hunk 3]
```diff
@@ -176,6 +179,7 @@ public function pull( $items ) {
 				update_post_meta( $new_post, 'dt_original_blog_id', (int) $this->site->blog_id );
 				update_post_meta( $new_post, 'dt_syndicate_time', time() );
 				update_post_meta( $new_post, 'dt_original_post_url', esc_url_raw( $post->link ) );
+				update_post_meta( $new_post, 'dt_original_post_parent', (int) $original_post_parent );
```
Comment (barryceelen) [inline includes/classes/InternalConnections/NetworkSiteConnection.php]: Same here: only store `$original_post_parent` if it is not `0`.

File: includes/classes/ExternalConnections/WordPressExternalConnection.php [hunk 4]
```diff
@@ -443,6 +445,7 @@ public function push( $post_id, $args = array() ) {
 			'distributor_original_site_name' => get_bloginfo( 'name' ),
 			'distributor_original_site_url'  => home_url(),
 			'distributor_original_post_url'  => get_permalink( $post_id ),
+			'distributor_original_post_parent'  => wp_get_post_parent_id( $post_id ),
```
Comment (barryceelen) [inline includes/classes/ExternalConnections/WordPressExternalConnection.php]: Only save `$original_post_parent` if it is not 0.

File: includes/classes/InternalConnections/NetworkSiteConnection.php [hunk 5]
```diff
@@ -85,6 +86,7 @@ public function push( $post_id, $args = array() ) {
 			update_post_meta( $new_post, 'dt_original_blog_id', (int) $original_blog_id );
 			update_post_meta( $new_post, 'dt_syndicate_time', time() );
 			update_post_meta( $new_post, 'dt_original_post_url', esc_url_raw( $original_post_url ) );
+			update_post_meta( $new_post, 'dt_original_post_parent', (int) $original_post_parent );
```
Comment (barryceelen) [inline includes/classes/InternalConnections/NetworkSiteConnection.php]: Same here: only store $original_post_parent if it is not 0.

**Comment by barryceelen** [comment]
Awesome, love keeping the parent post info preserved when distributing!

**Comment by tlovett1** [comment]
@sillybean do you have time to address @barryceelen's feedback? If not, we are happy to take over. Thank you!

**Comment by sillybean** [comment]
Please do! I’m leaving again today for another ten days.

**Comment by helen** [comment]
Could use some help getting this PR refreshed if we want it in 1.3.

**Pull request merged**